src/main/client_context.cpp

```diff
@@ -17,10 +20,14 @@ void ActiveQuery::reset() {
     timer = Timer();
 }
 
-ClientContext::ClientContext()
+ClientContext::ClientContext(Database* database)
     : numThreadsForExecution{std::thread::hardware_concurrency()},
       timeoutInMS{ClientContextConstants::TIMEOUT_IN_MS}, varLengthExtendMaxDepth{
-                                                              VAR_LENGTH_EXTEND_MAX_DEPTH} {}
+                                                              VAR_LENGTH_EXTEND_MAX_DEPTH} {
+    transactionContext = std::make_unique<TransactionContext>(database);
+}
+
+ClientContext::~ClientContext() {}
```

Comment:
If the deconstructor is empty, just let the compiler generates the default one.

Reply:
We are forward declaring. So cannot apply default constructor in header.